src/msg/simple/Pipe.cc

```diff
@@ -1342,6 +1344,7 @@ void Pipe::fault(bool onread)
     return;
   }
 
+  recv_reset();
   shutdown_socket();
```

Comment:
The other two sites were for shutdown cases -- so no worries if you leave state behind.  I can move to a new commit.